httpclient5/src/main/java/org/apache/hc/client5/http/impl/nio/MultihomeIOSessionRequester.java

```diff
@@ -129,7 +129,7 @@ public void cancelled() {
 
             void executeNext() {
                 final int index = attempt.getAndIncrement();
-                final InetSocketAddress remoteAddress = new InetSocketAddress(remoteAddresses[index], remoteEndpoint.getPort());
+                final InetSocketAddress remoteAddress = (InetSocketAddress) remoteAddresses.get(index);
```

Comment:
@arturobernalg I suppose this cast is no longer needed

Reply:
@ok2c I don't know what happened. I  just removed.